### PR TITLE
Config role

### LINKED ISSUE #399: Can no longer add roles to service accounts from apb service account.
**What happened**:
When running a provision of an existing apb, some of the commands started to fail.
The specific command was to add the edit role to a service account that is used to run the application.
**What you expected to happen**:
Continue be able to add a role in a namespace to the given service account.
**How to reproduce it**:
In an apb create a service account and then run
```
oc policy add-role-to-user edit system:serviceaccount:'{{namespace}}':'{{serviceAccount}}' -n '{{namespace}}'
```

**Result**
```
Error from server (Forbidden): User \"system:serviceaccount:myproject:apb-1b9524ae-ee29-4cea-85eb-aca2454c71e9\" cannot list rolebindings in project \"myproject\""
```

For reference this issue is discussed in the PR also:
https://github.com/openshift/ansible-service-broker/pull/393

### PR DESCRIPTION
**Pull request by djzager**
Make it possible to configure the role assigned to apb sandboxes.

**Which issue this PR fixes (This will close that issue when PR gets merged)**
fixes #399

### CODE REVIEW

File: pkg/app/config.go [hunk 1]
```diff
@@ -65,5 +65,12 @@ func validateConfig(c Config) error {
 		}
 		registryName[rc.Name] = true
 	}
+
+	switch role := c.Openshift.SandboxRole; role {
+	case apb.ApbRoleAdmin, apb.ApbRoleEdit:
+	default:
+		return fmt.Errorf("sandbox role is not valid - %v", role)
+	}
+
```
Comment (jmrodri) [inline pkg/app/config.go]: Exactly what I was trying to describe on your whiteboard. Glad it translated well :)

File: pkg/app/config_test.go [hunk 2]
```diff
@@ -37,6 +37,7 @@ func TestCreateConfig(t *testing.T) {
 	ft.AssertEqual(t, config.Openshift.CAFile, "", "")
 	ft.AssertEqual(t, config.Openshift.BearerTokenFile, "", "")
 	ft.AssertEqual(t, config.Openshift.PullPolicy, "IfNotPresent", "")
+	ft.AssertEqual(t, config.Openshift.SandboxRole, "edit", "")
```
Comment (jmrodri) [inline pkg/app/config_test.go]: ❤️ the test

File: pkg/app/config.go [hunk 3]
```diff
@@ -65,5 +65,12 @@ func validateConfig(c Config) error {
 		}
 		registryName[rc.Name] = true
 	}
+
+	switch role := c.Openshift.SandboxRole; role {
```
Comment (shawn-hurley) [inline pkg/app/config.go]: Can we make this a case insensitive match?

Comment (djzager) [reply pkg/app/config.go]: Is `switch role := strings.ToLower(c.Openshift.SandboxRole); role {` agreeable?

Comment (shawn-hurley) [reply pkg/app/config.go]: that is exactly what I was thinking

**Comment by jwmatthews** [comment]
I spoke to @djzager about the allowed roles, I would like to remove locking down supported roles and allow the administrator configuring the broker to specify whatever role they deem appropriate if overriding.

Let's default to "edit", yet allow Administrator to specify anything they want if overriding.

**Comment by djzager** [comment]
@shawn-hurley @jmrodri Massive change where I un-lock what roles can be used by the sandbox. Just wanted to give a heads up.

**Comment by djzager** [comment]
I haven't forgotten about your earlier comment @shawn-hurley and I'm updating catasb to provide a way to override the `sandbox_role`

File: docs/config.md [hunk 4]
```diff
@@ -147,8 +147,7 @@ registry:
 | ca_file | Location of the certificate authority file | N |
 | bearer_token_file | Location of bearer token to be used | N |
 | image_pull_policy | When to pull the image | Y |
-| sandbox_role | Role to give to apb sandbox environment. Supported values:
-`admin`, `edit` | Y |
+| sandbox_role | Role to give to apb sandbox environment. Supported values: `admin`, `edit` | Y |
```
Comment (jmrodri) [inline docs/config.md]: 👍

File: scripts/deploy.sh [hunk 5]
```diff
@@ -38,7 +39,8 @@ VARS="-p BROKER_IMAGE=${BROKER_IMAGE} \
   -p LAUNCH_APB_ON_BIND=${LAUNCH_APB_ON_BIND} \
   -p OUTPUT_REQUEST=${OUTPUT_REQUEST} \
   -p RECOVERY=${RECOVERY} \
-  -p REFRESH_INTERVAL=${REFRESH_INTERVAL}"
+  -p REFRESH_INTERVAL=${REFRESH_INTERVAL} \
+  -p SANDBOX_ROLE=${SANDBOX_ROLE}"
```
Comment (jmrodri) [inline scripts/deploy.sh]: 👍

File: templates/deploy-ansible-service-broker.template.yaml [hunk 6]
```diff
@@ -359,4 +360,10 @@ parameters:
   displayname: Cluster Authentication Host
   name: CLUSTER_AUTH_HOST
   value: ""
+
+- description: Role to use for APB Sandboxes
+  displayname: Role to use for APB Sandboxes
+  name: SANDBOX_ROLE
+  value: "edit"
```
Comment (jmrodri) [inline templates/deploy-ansible-service-broker.template.yaml]: 👍

File: docs/config.md [hunk 7]
```diff
@@ -147,7 +147,7 @@ registry:
 | ca_file | Location of the certificate authority file | N |
 | bearer_token_file | Location of bearer token to be used | N |
 | image_pull_policy | When to pull the image | Y |
-| sandbox_role | Role to give to apb sandbox environment. Supported values: `admin`, `edit` | Y |
+| sandbox_role | Role to give to apb sandbox environment | Y |
```
Comment (jmrodri) [inline docs/config.md]: Glad you remembered this change.

File: pkg/apb/types.go [hunk 8]
```diff
@@ -89,13 +89,6 @@ const (
 	// StateFailed - Failed job state
 	StateFailed State = "failed"
 
-	// The APB Roles that we support referencing when we create the APB Sandbox
-
-	// ApbRoleAdmin - Admin role
-	ApbRoleAdmin = "admin"
-	// ApbRoleEdit - Edit role
-	ApbRoleEdit = "edit"
-
```
Comment (jmrodri) [inline pkg/apb/types.go]: LGTM

**Pull request merged**